From 74fb16fafdbe6541e1476805827b21041f20049d Mon Sep 17 00:00:00 2001
From: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
Date: Thu, 25 Aug 2022 23:19:04 +0530
Subject: [PATCH] ath12k: Fix rx_per_chain_rssi_in_dbm is incorrect in
 htt_peer_stats

While printing the htt_peer_stats we printing the rx_per_chain_rssi_in_dbm
buffer by PRINT_ARRAY_TO_BUF define. This is common define for scnprintf in
%u:%u format. rx_per_chain_rssi_in_dbm is a s8 variable, but we printing
this by %u. So only getting this wrapper rounded values.

rx_per_chain_rssi_in_dbm[0] =  0:4294967250, 1:4294967254, 2:4294967251, 3:4294967250,
rx_per_chain_rssi_in_dbm[1] =  0:4294967252, 1:4294967253, 2:4294967252, 3:4294967252,
rx_per_chain_rssi_in_dbm[2] =  0:4294967256, 1:4294967256, 2:4294967257, 3:4294967256,
rx_per_chain_rssi_in_dbm[3] =  0:4294967248, 1:4294967248, 2:4294967247, 3:4294967248,

Now changed the code for printing the rx_per_chain_rssi_in_dbm buffer in %d.

Signed-off-by: Sivashankari Madhavan <quic_sivamadh@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c | 12 ++++++++----
 1 file changed, 8 insertions(+), 4 deletions(-)

diff --git a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
index 4277942..4ba769a 100644
--- a/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
+++ b/drivers/net/wireless/ath/ath12k/debugfs_htt_stats.c
@@ -930,7 +930,7 @@ static inline void htt_print_rx_peer_rate_stats_tlv(const void *tag_buf,
 	u8 *buf = stats_req->buf;
 	u32 len = stats_req->buf_len;
 	u32 buf_len = ATH12K_HTT_STATS_BUF_SIZE;
-	u8 j;
+	u8 i, j;
 	char *rssi_chain[HTT_RX_PEER_STATS_NUM_SPATIAL_STREAMS] = {NULL};
 	char *rx_gi[HTT_RX_PEER_STATS_NUM_GI_COUNTERS] = {NULL};
 
@@ -1021,10 +1021,14 @@ static inline void htt_print_rx_peer_rate_stats_tlv(const void *tag_buf,
 	for (j = 0; j < HTT_RX_PEER_STATS_NUM_SPATIAL_STREAMS; j++) {
 		len += scnprintf(buf + len, buf_len - len,
 				 "\nrx_per_chain_rssi_in_dbm[%u] = ", j);
-		PRINT_ARRAY_TO_BUF(buf, len, htt_stats_buf->rx_per_chain_rssi_in_dbm[j],
-				   NULL, HTT_RX_PEER_STATS_NUM_BW_COUNTERS, "\n");
+		for (i = 0; i < HTT_RX_PDEV_STATS_NUM_BW_COUNTERS; i++)
+			len += scnprintf(buf + len,
+					 buf_len - len,
+					 " %u:%d,",
+					 i,
+					 htt_stats_buf->rx_per_chain_rssi_in_dbm[j][i]);
+		len += scnprintf(buf + len, buf_len - len, "\n");
 	}
-
 	stats_req->buf_len = len;
 
 fail:
-- 
2.17.1

